firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/RemoteConfigDeferredProxy.java

```diff
@@ -0,0 +1,41 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.crashlytics;
+
+import com.google.firebase.crashlytics.internal.metadata.UserMetadata;
+import com.google.firebase.inject.Deferred;
+import com.google.firebase.remoteconfig.interop.FirebaseRemoteConfigInterop;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class RemoteConfigDeferredProxy {
+  private final Deferred<FirebaseRemoteConfigInterop> remoteConfigInteropDeferred;
+  private final AtomicReference<UserMetadata> userMetadata = new AtomicReference<>(null);
+
+  public RemoteConfigDeferredProxy(
+      Deferred<FirebaseRemoteConfigInterop> remoteConfigInteropDeferred) {
+    this.remoteConfigInteropDeferred = remoteConfigInteropDeferred;
+  }
+
+  public void setupListener(UserMetadata userMetadata) {
+    this.userMetadata.set(userMetadata);
+    remoteConfigInteropDeferred.whenAvailable(
+        remoteConfigInteropProvider -> {
+          CrashlyticsRemoteConfigListener listener =
+              new CrashlyticsRemoteConfigListener(this.userMetadata.get());
+          FirebaseRemoteConfigInterop interop = remoteConfigInteropProvider.get();
+          interop.registerRolloutsStateSubscriber("crashlytics", listener);
```

Comment:
Could we add a debug log here so we know when it's subscribed to help with debugging issues?

Reply:
Could be something like, "Registering RemoteConfig Rollouts subscriber"